source/testing/testing-components.md

```diff
@@ -25,74 +25,122 @@ export default Component.extend({
 
 ```tests/integration/components/pretty-color-test.js
-import { moduleForComponent, test } from 'ember-qunit';
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
 
-moduleForComponent('pretty-color', 'Integration | Component | pretty color', {
-  integration: true
+module('Integration | Component | pretty color', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('should change colors', async function(assert) {
+    assert.expect(1);
+
+    // set the outer context to red
+    this.set('colorValue', 'red');
+
+    await render(hbs`{{pretty-color name=colorValue}}`);
+
+    assert.equal(this.element.querySelector('div').getAttribute('style'), 'color: red', 'starts as red');
+  });
 });
 
-test('should change colors', function(assert) {
-  assert.expect(2);
+```
+
+Each test can use the `render()`
+function imported from the `@ember/test-helpers` package to create a new instance of the component by declaring
+the component in template syntax, as we would in our application.
+Also notice, the keyword `await` in front of the call to `render`. It allows the test which we marked as `async` earlier to wait for any asynchronous behaviour to complete before any code which is written below that will be executed.
+In this case our first assertion will correctly execute after the component has fully rendered.
+
+Next we can test that changing the component's `name` property updates the
+component's `style` attribute and is reflected in the rendered HTML:
+
+```tests/integration/components/pretty-color-test.js
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
 
-  // set the outer context to red
-  this.set('colorValue', 'red');
+module('Integration | Component | pretty color', function(hooks) {
+  setupRenderingTest(hooks);
 
-  this.render(hbs`{{pretty-color name=colorValue}}`);
+  test('it renders', async function(assert) {
+    assert.expect(2);
 
-  assert.equal(this.$('div').attr('style'), 'color: red', 'starts as red');
+    // set the outer context to red
+    this.set('colorValue', 'red');
 
-  this.set('colorValue', 'blue');
+    await render(hbs`{{pretty-color name=colorValue}}`);
 
-  assert.equal(this.$('div').attr('style'), 'color: blue', 'updates to blue');
+    assert.equal(this.element.querySelector('div').getAttribute('style'), 'color: red', 'starts as red');
+
+    this.set('colorValue', 'blue');
```

Comment:
@rwjblue I think this will be problematic once we do async rendering, right? would it technically need an `await settled()` then? do we need to adjust this or is this fine for now?

Reply:
Yes, indeed.  The only reason `this.set` works is because it internally "run wraps" things and forces the rendering to happen.

I'm fine with this for now because we will still have to support forced-sync rendering (specifically in order to avoid breaking tests), we can add the `await settled()` (or a new `await rendered()`) once it is possible to do the `set` without run-wrapping.